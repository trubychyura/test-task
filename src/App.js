import { useEffect, useState } from 'react';
import { Button, Container, Table } from 'react-bootstrap';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { loadData } from './redux/products';

import Modal from './components/Modal';
import Product from './components/Product';

const StyledContainer = styled(Container)`
  padding: 20px 10px;
`;

const StyledTable = styled(Table)`
  text-align: center;
  th {
    font-size: 40px;
  }

  tbody tr {
    cursor: pointer;
  }
`;

function App({ loadData, products }) {
  useEffect(() => {
    loadData();
    console.log('loading...');
  }, [loadData]);

  return (
    <StyledContainer>
      <Modal />
      <StyledTable bordered striped hover>
        <thead>
          <tr>
            <th>List of products</th>
          </tr>
        </thead>
        <tbody>
          {products.map(({ name, asin, img, comments = [] }) => {
            const data = {
              name,
              asin,
              img,
              comments,
            };
            return <Product data={data} key={asin} />;
          })}
        </tbody>
      </StyledTable>
    </StyledContainer>
  );
}

const mapStateToProps = ({ products }) => ({ products: products.products });

export default connect(mapStateToProps, {
  loadData,
})(App);
