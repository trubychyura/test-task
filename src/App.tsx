import { useEffect, FC } from 'react';
import { connect } from 'react-redux';
import { loadData } from './redux/products';
import { StyledContainer, StyledTable } from './styled/StyledApp';
import { HashRouter as Router, Route } from 'react-router-dom';

import Modal from './components/Modal';
import Product from './components/Product';
import { IProduct, IState } from './types';

type AppProps = {
  products: IProduct[];
  loadData: () => void;
};

const App: FC<AppProps> = ({ loadData, products }) => {
  useEffect(() => {
    loadData();
  }, [loadData]);

  return (
    <StyledContainer>
      <Router>
        <Route path='/'>
          <StyledTable bordered striped hover>
            <thead>
              <tr>
                <th>List of products</th>
              </tr>
            </thead>
            <tbody>
              {products.map(({ name, asin }) => (
                <Product id={asin} name={name} key={asin} />
              ))}
            </tbody>
          </StyledTable>
        </Route>
        <Route path='/:id'>
          <Modal />
        </Route>
      </Router>
    </StyledContainer>
  );
};

export default connect(({ products }: IState) => ({ products }), {
  loadData,
})(App);
