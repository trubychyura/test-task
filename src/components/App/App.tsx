import { FC } from 'react';
import { HashRouter as Router, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { Modal, Home } from '..';

import theme from '../../styled/theme';

const App: FC = () => (
  <ThemeProvider theme={theme}>
    <Router>
      <Route path='/'>
        <Home />
      </Route>
      <Route path='/:id'>
        <Modal />
      </Route>
    </Router>
  </ThemeProvider>
);

export default App;
