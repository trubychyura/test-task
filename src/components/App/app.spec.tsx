import { shallow } from 'enzyme';

import App from './App';

it('should render App component', () => {
  expect(shallow(<App />)).toMatchSnapshot();
});
