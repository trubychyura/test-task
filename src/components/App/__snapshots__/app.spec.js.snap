// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App component 1`] = `
<Component
  theme={
    Object {
      "color": Object {
        "gold": "gold",
        "silver": "silver",
      },
      "media": Object {
        "lg": "1000px",
        "md": "580px",
      },
    }
  }
>
  <HashRouter>
    <Route
      path="/"
    >
      <Connect(Home) />
    </Route>
    <Route
      path="/:id"
    >
      <Connect(Modal) />
    </Route>
  </HashRouter>
</Component>
`;
